### ISSUE TITLE
Indentation stays incorrect

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#3247]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This can happen in combination with #3246, making it really annoying.

For `order by` clauses with several fields, the indentation seems to be ambiguous: It can be normally indented (one indent more than `order`), or aligned with the previous field.
If the indentation of the second field is aligned, but the `order by` clause is wrong, `sqlfluff fix` switches between to indentations, but cannot fix the whole statement.

It seems that this is related to some interference between L003 and L039. I can work around this by excluding L039, but this can only be temporary.

As an example, take this:
```
begin
  begin
    select Blah
    from Blubb
 order by Field1 desc,
             Field2 desc
  end
end
```
Note that `Field2` would be correctly aligned with `Field1` if `order by Field1` were correctly indented.

### Expected Behaviour

`order by` should be indented correctly. Then, I'd prefer to have `Field2` just one more indent than the previous line, but it is also acceptable if the fields are aligned.

### Observed Behaviour
After `sqlfluff fix`:
```
begin
  begin
    select Blah
    from Blubb
 order by Field1 desc,
         Field2 desc
  end
end
```
`Field2` is neither correctly indented nor aligned.

After the next `sqlfluff fix`, we're back at the beginning:
```
begin
  begin
    select Blah
    from Blubb
 order by Field1 desc,
             Field2 desc
  end
end
```

### Other observations
Correctly indenting `Field2` manually also solves this problem.
Removing just one space character from `Field2`'s indentation will take two iterations, but then the whole statement is correctly formatted and stable. Maybe it is also relevant that the broken indentations are 17 (aligned with previous fields) and 13, both odd numbers?

### How to reproduce

See above

### Dialect

T-SQL

### Version

Latest `main`

### Configuration

```
[sqlfluff:rules]
tab_space_size = 2
indent_unit = space
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
@fdw: This is the output I'm seeing on the latest `main` (commit `77e283044c2a7d83db22f9df818cea96f8d2e0cb`). Please test it again and let us know if it's working correctly.
```
begin
  begin
    select Blah
    from Blubb
    order by Field1 desc,
             Field2 desc
  end
end
```

**Comment by fdw** [comment]
Unfortunately, `pip` doesn't write out which exact commit I'm using. I guess it must be `ddbd330f2e45ae4ea99273cb5cd86265a1692962`.

Using the exact code I shared above, I still see the same oscillating pattern with `Field2 desc` indented either 9 or 13 spaces.

My `.sqlfluff` is empty apart from defining the dialect (`tsql`) and the two lines about indentation I shared. I'm on Python 3.8.10, `sqlfluff` version is `0.13.1` (but again, installed with pip from git).

**Comment by alanmcruickshank** [comment]
I think this might be an issue with hanging indents which is resolved in #3942. On that branch, the query is fixed to:

```sql
begin
    begin
        select Blah
        from Blubb
        order by
            Field1 desc,
            Field2 desc
    end
end
```
